l10n_it_invoices_data_communication/models/communication.py

```diff
@@ -43,6 +44,16 @@ def check_normalized_string(value):
     return normalized
 
 
+def normalize_encode(value, form='NFKD', encoding='ascii', errors='replace'):
+    try:
+        # Soluzione proposta in PR #1505
+        # res = value.encode('latin', 'replace').decode('latin')
+        res = unicodedata.normalize(form, value).encode(encoding=encoding, errors=errors)
```

Comment:
Perchè `unicodedata.normalize` invece di `encode('latin', 'replace').decode('latin')`?
Inoltre mi pare che qui butti via tutto ciò che non è ascii, mentre i caratteri latin dobbiamo tenerli.

Reply:
Perché con` encode('latin', 'replace').decode('latin')` non vengono rimossi del tutto i simboli pre-codificati in UTF-8 presenti nella stringa, ma vengono riportati lettera per lettera nel risultato.

![image](https://user-images.githubusercontent.com/16492632/67564481-509ea100-f723-11e9-8584-fb181c63e4bf.png)
 
L'ultima freccia rossa in basso mostra che i caratteri _\xdf_ sono rimasti nel risultato invece di essere convertiti o sostituiti. Tramite `unicodedata.normalize(form, value).encode(encoding=encoding, errors=errors)` invece essi vengono riconosciuti e interpretati come unico simbolo secondo la tabella dei simboli di UTF-8. Successivamente nella `encode(...)`, il simbolo viene mappato con il corrispondente simbolo nella tabella ASCII e questa è una forzatura doverosa, altrimenti mappdando in LATIN non viene rimosso!

![image](https://user-images.githubusercontent.com/16492632/67564934-6b254a00-f724-11e9-8bc7-f444cc461154.png)

Questo è quando è emerso dalle prove che ho effettuato su Odoo. Ho riportato degli esempio esempi su terminale Python per semplicità, ma i risultati nel file XML risultato è lo stesso.

Reply:
la soluzione che ho adottato è ispirata ad un post che riporto di seguito per completezza
https://stackoverflow.com/questions/1207457/convert-a-unicode-string-to-a-string-in-python-containing-extra-symbols

Reply:
Mi puoi riportare il carattere non latin che non verrebbe rimosso?

Reply:
Ah vedo ora il commento https://github.com/OCA/l10n-italy/pull/1509#discussion_r339069371 quindi @Byloth ti ha già spiegato meglio